Study dimension elements endpoint fails

The request `POST https://transmart-dev.thehyve.net/v2/dimensions/study/elements` (e.g., on the data table page of Glowing Bear) fails because of:
```
Jun 30 14:54:32 transmart-dev start[2465]: 2020-06-30 14:54:32.141  WARN --- [nio-8080-exec-8] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 0, SQLState: 42P01
Jun 30 14:54:32 transmart-dev start[2465]: 2020-06-30 14:54:32.142 ERROR --- [nio-8080-exec-8] o.h.engine.jdbc.spi.SqlExceptionHelper   : ERROR: relation "i2b2demodata.study_trial_visit_dimension" does not exist
Jun 30 14:54:32 transmart-dev start[2465]:   Position: 579
Jun 30 14:54:32 transmart-dev start[2465]: 2020-06-30 14:54:32.143  INFO --- [nio-8080-exec-8] o.t.rest.http.BusinessExceptionResolver  : Asked BusinessExceptionResolver to resolve exception from handler org.grails.web.mapping.mvc.GrailsControllerUrlMappingInfo@d353e7c
Jun 30 14:54:32 transmart-dev start[2465]: org.hibernate.exception.SQLGrammarException: could not extract ResultSet
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:106)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:42)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:111)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:97)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.extract(ResultSetReturnImpl.java:79)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.loader.Loader.getResultSet(Loader.java:2122)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:1905)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:1881)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.loader.Loader.scroll(Loader.java:2698)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.loader.criteria.CriteriaLoader.scroll(CriteriaLoader.java:104)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.internal.SessionImpl.scroll(SessionImpl.java:1740)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.internal.CriteriaImpl.scroll(CriteriaImpl.java:377)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.transmartproject.db.clinical.AbstractDataResourceService.getIterable(AbstractDataResourceService.groovy:76)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.transmartproject.db.clinical.MultidimensionalDataResourceService.getDimensionElements(MultidimensionalDataResourceService.groovy:317)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.transmartproject.db.clinical.MultidimensionalDataResourceService.getDimensionElements(MultidimensionalDataResourceService.groovy)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.transmartproject.core.multidimquery.MultiDimensionalDataResource$getDimensionElements.call(Unknown Source)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.transmartproject.core.multidimquery.MultiDimensionalDataResource$getDimensionElements.call(Unknown Source)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.transmartproject.rest.DimensionController.listElements(DimensionController.groovy:134)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.transmartproject.rest.DimensionController.listElements(DimensionController.groovy)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.grails.core.DefaultGrailsControllerClass$MethodHandleInvoker.invoke(DefaultGrailsControllerClass.java:223)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.grails.core.DefaultGrailsControllerClass.invoke(DefaultGrailsControllerClass.java:188)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.grails.web.mapping.mvc.UrlMappingsInfoHandlerAdapter.handle(UrlMappingsInfoHandlerAdapter.groovy:90)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
Jun 30 14:54:32 transmart-dev start[2465]:         at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
Jun 30 14:54:32 transmart-dev start[2465]:         at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:54)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticatedActionsFilter.doFilter(KeycloakAuthenticatedActionsFilter.java:57)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakSecurityContextRequestFilter.doFilter(KeycloakSecurityContextRequestFilter.java:61)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakPreAuthActionsFilter.doFilter(KeycloakPreAuthActionsFilter.java:96)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticationProcessingFilter.successfulAuthentication(KeycloakAuthenticationProcessingFilter.java:214)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:240)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.boot.actuate.trace.WebRequestTraceFilter.doFilterInternal(WebRequestTraceFilter.java:111)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:77)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticatedActionsFilter.doFilter(KeycloakAuthenticatedActionsFilter.java:74)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakSecurityContextRequestFilter.doFilter(KeycloakSecurityContextRequestFilter.java:92)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticationProcessingFilter.successfulAuthentication(KeycloakAuthenticationProcessingFilter.java:214)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:240)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.springsecurity.filter.KeycloakPreAuthActionsFilter.doFilter(KeycloakPreAuthActionsFilter.java:96)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:66)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:96)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:100)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.tomcat.AbstractAuthenticatedActionsValve.invoke(AbstractAuthenticatedActionsValve.java:67)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.keycloak.adapters.tomcat.AbstractKeycloakAuthenticatorValve.invoke(AbstractKeycloakAuthenticatorValve.java:181)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1627)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
Jun 30 14:54:32 transmart-dev start[2465]:         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Jun 30 14:54:32 transmart-dev start[2465]:         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
Jun 30 14:54:32 transmart-dev start[2465]:         at java.lang.Thread.run(Thread.java:748)
Jun 30 14:54:32 transmart-dev start[2465]: Caused by: org.postgresql.util.PSQLException: ERROR: relation "i2b2demodata.study_trial_visit_dimension" does not exist
Jun 30 14:54:32 transmart-dev start[2465]:   Position: 579
Jun 30 14:54:32 transmart-dev start[2465]:         at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2532)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2267)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:312)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:448)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:369)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:153)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:103)
Jun 30 14:54:32 transmart-dev start[2465]:         at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
Jun 30 14:54:32 transmart-dev start[2465]:         at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
Jun 30 14:54:32 transmart-dev start[2465]:         at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.extract(ResultSetReturnImpl.java:70)
Jun 30 14:54:32 transmart-dev start[2465]:         ... 128 common frames omitted
Jun 30 14:54:32 transmart-dev start[2465]: 2020-06-30 14:54:32.163 TRACE --- [nio-8080-exec-8] o.t.db.log.AccessLogService              : {"username":"xxxxxxxx","event":"error","eventMessage":"{\"ip\":\"xxxxxxx\",\"action\":\"index\",\"duration\":39,\"status\":500,\"message\":\"could not extract ResultSet\",\"body\":{\"constraint\":{\"type\":\"true\"}}}","requestURL":"/v2/dimensions/study/elements","accessTime":"2020-06-30T14:54:32"}
```

--- a/transmart-core-db-tests/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServiceSpec.groovy
+++ b/transmart-core-db-tests/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServiceSpec.groovy
@@ -475,7 +475,7 @@ class QueryServiceSpec extends Specification {
         setupHypercubeData()
         DimensionImpl dimension = DimensionImpl.TRIAL_VISIT
         Constraint constraint = new StudyNameConstraint(hypercubeTestData.clinicalData.multidimsStudy.studyId )
-        def expectedResults = hypercubeTestData.clinicalData.multidimsStudy.trialVisits as Set<TrialVisit>
+        def expectedResults = hypercubeTestData.clinicalData.trialVisits as Set<TrialVisit>
 
         when:"I query for all trial visits for a constraint"
         def trialVisits = multiDimService.getDimensionElements(dimension.name, constraint, accessLevelTestData.users[1]).collect {

--- a/transmart-core-db-tests/src/main/groovy/org/transmartproject/db/dataquery/clinical/ClinicalTestData.groovy
+++ b/transmart-core-db-tests/src/main/groovy/org/transmartproject/db/dataquery/clinical/ClinicalTestData.groovy
@@ -60,6 +60,7 @@ class ClinicalTestData {
     Study                   ehrStudy
     List<ObservationFact>   ehrClinicalFacts
     Study                   multidimsStudy
+    List<TrialVisit>        trialVisits
     List<ObservationFact>   multidimsClinicalFacts
     List<ModifierDimensionCoreDb> modifierDimensions
     ModifierDimension       doseDimension
@@ -126,7 +127,13 @@ class ClinicalTestData {
 
         def multidimsStudy = StudyTestData.createStudy "multidimensional study", ["patient", "concept", "study", "visit", "trial visit",
                                                                "start time", "end time", "location", "provider"], true
-        def multidimsClinicalFacts = createMultidimensionalFacts(conceptDims[5..6], visits[3..5], multidimsStudy, observationStartDates, observationEndDates,
+
+        def trialVisits = [
+                createTrialVisit('day', 0, 'baseline', multidimsStudy),
+                createTrialVisit('day', 10, 'after 10 days', multidimsStudy),
+                createTrialVisit('day', 35, 'after 5 weeks', multidimsStudy)]
+
+        def multidimsClinicalFacts = createMultidimensionalFacts(conceptDims[5..6], visits[3..5], trialVisits, observationStartDates, observationEndDates,
                 locations, providers)
 
         def longitudinalStudy = StudyTestData.createStudy "longitudinal study", ["patient", "concept", "trial visit", "study"]
@@ -167,7 +174,8 @@ class ClinicalTestData {
                 tissueTypeDimension: tissueTypeDimension.dimension as ModifierDimension,
                 ehrStudy: ehrStudy,
                 ehrClinicalFacts: ehrClinicalFacts,
-                multidimsStudy:multidimsStudy,
+                multidimsStudy: multidimsStudy,
+                trialVisits: trialVisits,
                 multidimsClinicalFacts: multidimsClinicalFacts,
                 modifierDimensions: [modifierDimensions.left]
         )
@@ -446,15 +454,11 @@ class ClinicalTestData {
     }
 
 
-    static List<ObservationFact> createMultidimensionalFacts (List<ConceptDimension> concept, List<VisitDimension> visits,
-                                                         Study study, List<Date> startDates, List<Date> endDates,
-                                                         List<String> locations, List<String> providers){
+    static List<ObservationFact> createMultidimensionalFacts (
+            List<ConceptDimension> concept, List<VisitDimension> visits,
+            List<TrialVisit> trialVisits, List<Date> startDates, List<Date> endDates,
+            List<String> locations, List<String> providers){
 
-        // These trialVisits don't really make sense, since different trial visits share the same visit, but it works
-        // fine for testing
-        def trialVisits = [createTrialVisit('day', 0, 'baseline', study),
-                           createTrialVisit('day', 10, 'after 10 days', study),
-                           createTrialVisit('day', 35, 'after 5 weeks', study)]
         def istartDates = peekingIterator(cycle(startDates))
         def iendDates = peekingIterator(cycle(endDates))
         def ilocations = peekingIterator(cycle(locations))
@@ -499,7 +503,6 @@ class ClinicalTestData {
                 relTimeLabel: relTimeLabel,
         )
         tv.study = study
-        study.addToTrialVisits(tv)
         tv
     }
 

--- a/transmart-core-db/grails-app/domain/org/transmartproject/db/i2b2data/Study.groovy
+++ b/transmart-core-db/grails-app/domain/org/transmartproject/db/i2b2data/Study.groovy
@@ -55,8 +55,7 @@ class Study implements MDStudy {
     }
 
     static hasMany = [
-        trialVisits: TrialVisit,
-        dimensionDescriptions: DimensionDescription,
+        dimensionDescriptions: DimensionDescription
     ]
 
     static mapping = {

--- a/transmart-core-db/grails-app/domain/org/transmartproject/db/i2b2data/TrialVisit.groovy
+++ b/transmart-core-db/grails-app/domain/org/transmartproject/db/i2b2data/TrialVisit.groovy
@@ -2,6 +2,9 @@
 
 package org.transmartproject.db.i2b2data
 
+import groovy.transform.EqualsAndHashCode
+
+@EqualsAndHashCode(includes = ['id'])
 class TrialVisit implements org.transmartproject.core.multidimquery.TrialVisit {
 
     String relTimeUnit
@@ -19,7 +22,7 @@ class TrialVisit implements org.transmartproject.core.multidimquery.TrialVisit {
     ]
 
     static belongsTo = [
-            study: Study
+        study: Study
     ]
 
     static mapping = {

--- a/transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServicePgSpec.groovy
+++ b/transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServicePgSpec.groovy
@@ -18,6 +18,7 @@ import org.transmartproject.core.multidimquery.query.Operator
 import org.transmartproject.core.multidimquery.query.PatientSetConstraint
 import org.transmartproject.core.multidimquery.query.StudyNameConstraint
 import org.transmartproject.core.multidimquery.query.TimeConstraint
+import org.transmartproject.core.multidimquery.query.TrueConstraint
 import org.transmartproject.core.multidimquery.query.Type
 import org.transmartproject.core.multidimquery.query.ValueConstraint
 import org.transmartproject.core.querytool.QueryResult
@@ -152,6 +153,16 @@ class QueryServicePgSpec extends Specification {
         then: "List of all visits matching the constraints is returned"
         visits.size() == expectedResult.size()
         visits.collect { it.encounterIds['VISIT_ID'] }.sort() == expectedResult.collect { it.encounterIds['VISIT_ID']}.sort()
+    }
+
+    void "test query all observations"() {
+        def user = User.findByUsername('test-public-user-1')
+
+        when: "I enumerate all observations"
+        def results = multiDimService.retrieveClinicalData(new TrueConstraint(), user).asList()
+
+        then: "No errors occur and the observation count is non-zero"
+        results.size() > 0
     }
 
     void "test patient set query"() {

--- a/transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServicePgSpec.groovy
+++ b/transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/QueryServicePgSpec.groovy
@@ -165,6 +165,24 @@ class QueryServicePgSpec extends Specification {
         results.size() > 0
     }
 
+    void "test query for 'study' dimension elements"() {
+        def user = User.findByUsername('test-public-user-1')
+        DimensionImpl dimension = STUDY
+
+        Constraint constraint = new TrueConstraint()
+        def results = multiDimService.retrieveClinicalData(constraint, user).asList()
+        def expectedResult = results.collect { it[STUDY] }.findAll { it } as Set
+
+        when: "I query for all studies for a constraint"
+        def studies = multiDimService.getDimensionElements(dimension.name, constraint, user).collect {
+            dimension.asSerializable(it)
+        }
+
+        then: "List of all studies matching the constraints is returned"
+        studies.size() == expectedResult.size()
+        studies.collect { it.name }.sort() == expectedResult.collect { it.studyId }.sort()
+    }
+
     void "test patient set query"() {
         def user = User.findByUsername('test-public-user-1')
 

--- a/transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/SurveyTableViewSpec.groovy
+++ b/transmart-core-db/src/integration-test/groovy/org/transmartproject/db/multidimquery/SurveyTableViewSpec.groovy
@@ -94,11 +94,11 @@ class SurveyTableViewSpec extends Specification {
         secondSubjRow
         secondSubjRow[columns[0]] == '2'
         secondSubjRow[columns[1]] == Date.parse('yyyy-MM-dd', '1986-10-22', UTC)
-        secondSubjRow[columns[2]] == Date.parse(DATE_TIME_FORMAT, '2010-12-16 20:23:15')
+        secondSubjRow[columns[2]] == Date.parse(DATE_TIME_FORMAT, '2010-12-16 20:23:15', UTC)
         secondSubjRow[columns[3]] == Date.parse(DATE_TIME_FORMAT, '2001-09-01 05:30:05', UTC)
-        secondSubjRow[columns[4]] == Date.parse(DATE_TIME_FORMAT, '2010-12-16 20:23:15')
+        secondSubjRow[columns[4]] == Date.parse(DATE_TIME_FORMAT, '2010-12-16 20:23:15', UTC)
         that secondSubjRow[columns[5]] as String, containsString('Dostoyevsky')
-        secondSubjRow[columns[6]] == Date.parse(DATE_TIME_FORMAT, '2016-03-21 10:36:01')
+        secondSubjRow[columns[6]] == Date.parse(DATE_TIME_FORMAT, '2016-03-21 10:36:01', UTC)
         secondSubjRow[columns[7]] == -2
         secondSubjRow[columns[8]] == null
         secondSubjRow[columns[13]] == '3'
@@ -106,11 +106,11 @@ class SurveyTableViewSpec extends Specification {
         def firstSubjRow = rows.find { row ->  row[columns[0]] == '1' }
         firstSubjRow[columns[0]] == '1'
         firstSubjRow[columns[1]] == Date.parse('yyyy-MM-dd', '1980-08-12', UTC)
-        firstSubjRow[columns[2]] == Date.parse(DATE_TIME_FORMAT, '2015-11-14 19:05:00')
+        firstSubjRow[columns[2]] == Date.parse(DATE_TIME_FORMAT, '2015-11-14 19:05:00', UTC)
         firstSubjRow[columns[3]] == null
         firstSubjRow[columns[4]] == null
         that firstSubjRow[columns[5]] as String, containsString('Hofstadter')
-        firstSubjRow[columns[6]] == Date.parse(DATE_TIME_FORMAT, '2016-03-21 10:36:01')
+        firstSubjRow[columns[6]] == Date.parse(DATE_TIME_FORMAT, '2016-03-21 10:36:01', UTC)
         firstSubjRow[columns[7]] == 2
         firstSubjRow[columns[8]] == null
         firstSubjRow[columns[13]] == 10
@@ -173,9 +173,9 @@ class SurveyTableViewSpec extends Specification {
         firstSubjRow
         firstSubjRow[columns[0]] == '1'
         firstSubjRow[columns[1]] == 'Description about subject 1'
-        firstSubjRow[columns[2]] == Date.parse(DATE_TIME_FORMAT, '2016-03-21 10:36:01')
+        firstSubjRow[columns[2]] == Date.parse(DATE_TIME_FORMAT, '2016-03-21 10:36:01', UTC)
         firstSubjRow[columns[3]] == -1
-        firstSubjRow[columns[4]] == Date.parse(DATE_TIME_FORMAT, '2005-05-24 13:40:00')
+        firstSubjRow[columns[4]] == Date.parse(DATE_TIME_FORMAT, '2005-05-24 13:40:00', UTC)
 
         def secondSubjRow = rows.find { row ->  row[columns[0]] == '2' }
         secondSubjRow
@@ -183,7 +183,7 @@ class SurveyTableViewSpec extends Specification {
         secondSubjRow[columns[1]] == 'No description'
         secondSubjRow[columns[2]] == null
         secondSubjRow[columns[3]] <=> 169 == 0
-        secondSubjRow[columns[4]] == Date.parse(DATE_TIME_FORMAT, '2004-08-27 10:45:32')
+        secondSubjRow[columns[4]] == Date.parse(DATE_TIME_FORMAT, '2004-08-27 10:45:32', UTC)
 
         when: 'do not include MeasurementDateColumn'
         includeMeasurementDateColumns = false

--- a/transmart-core-db/src/main/groovy/org/transmartproject/db/multidimquery/DimensionImpl.groovy
+++ b/transmart-core-db/src/main/groovy/org/transmartproject/db/multidimquery/DimensionImpl.groovy
@@ -684,14 +684,18 @@ class StudyDimension extends I2b2Dimension<MDStudy, String> implements Composite
     @Override List<MDStudy> doResolveElements(List<String> elementKeys) {
         resolveWithInQuery({ -> I2b2Study.createCriteria() as HibernateCriteriaBuilder}, elementKeys, 'studyId')
     }
+
     @Override
     DetachedCriteria selectDimensionElements(DetachedCriteria criteria) {
         criteria.add(HibernateCriteriaQueryBuilder.defaultModifierCriterion)
         criteria.setProjection(Projections.property('trialVisit'))
 
+        def trialVisitCriteria = DetachedCriteria.forClass(TrialVisit, 'trialVisits')
+        trialVisitCriteria.add(Subqueries.propertyIn('id', criteria))
+        trialVisitCriteria.setProjection(Projections.property('study'))
+
         def dimensionCriteria = DetachedCriteria.forClass(I2b2Study, 'study')
-        dimensionCriteria.createAlias('trialVisits', 'trialVisits')
-        dimensionCriteria.add(Subqueries.propertyIn('trialVisits.id', criteria))
+        dimensionCriteria.add(Subqueries.propertyIn('id', trialVisitCriteria))
         dimensionCriteria
     }
 

--- a/transmart-data/test_studies/MULTI-DWH/i2b2demodata/observation_fact.tsv
+++ b/transmart-data/test_studies/MULTI-DWH/i2b2demodata/observation_fact.tsv
@@ -7,9 +7,9 @@ encounter_num	patient_num	concept_cd	provider_id	start_date	modifier_cd	instance
 -1	5	VAR_bloodpressure	@		@	5	0	N	E	70.2													
 -1	6	VAR_bloodpressure	@		@	6	0	N	E	81.4													
 -1	8	VAR_bloodpressure	@		@	7	0	N	E	66.6													
--1	1	VAR_bloodpressure	@		POS	0	0	T	Standing														
--1	5	VAR_bloodpressure	@		POS	1	0	T	Standing														
--1	6	VAR_bloodpressure	@		POS	2	0	T	Standing														
+1	1	VAR_bloodpressure	@		POS	0	0	T	Standing														
+2	5	VAR_bloodpressure	@		POS	1	0	T	Standing														
+3	6	VAR_bloodpressure	@		POS	2	0	T	Standing														
 -1	8	VAR_bloodpressure	@		POS	3	0	T	Standing														
 -1	1	VAR_bloodpressure	@		POS	4	0	T	Sitting														
 -1	5	VAR_bloodpressure	@		POS	5	0	T	Sitting														

--- a/transmart-rest-api/src/func-test/groovy/org/transmartproject/rest/data/AccessPolicyTestData.groovy
+++ b/transmart-rest-api/src/func-test/groovy/org/transmartproject/rest/data/AccessPolicyTestData.groovy
@@ -1,6 +1,6 @@
 package org.transmartproject.rest.data
 
-import grails.transaction.Transactional
+import grails.gorm.transactions.Transactional
 import groovy.util.logging.Slf4j
 import org.hibernate.criterion.DetachedCriteria
 import org.hibernate.criterion.Restrictions
@@ -23,7 +23,7 @@ import static java.util.Objects.requireNonNull
 import static org.transmartproject.core.ontology.OntologyTerm.VisualAttributes.CATEGORICAL
 
 @Slf4j
-class AccessPolicyTestData extends org.transmartproject.rest.data.TestData {
+class AccessPolicyTestData extends TestData {
 
     @Transactional
     void createTestData() {
@@ -74,24 +74,24 @@ class AccessPolicyTestData extends org.transmartproject.rest.data.TestData {
     protected static final Set<String> defaultDimensions =
             ['study', 'concept', 'patient', 'start time', 'end time', 'trial visit'] as Set<String>
 
-    protected List<TrialVisit> createTestStudyAndTrialVisits(String studyId, boolean isPublic, List<String> trialVisitLabels) {
+    protected static List<TrialVisit> createTestStudyAndTrialVisits(String studyId, boolean isPublic, List<String> trialVisitLabels) {
         log.info "Creating test study: ${studyId}"
+        List<TrialVisit> trialVisits = []
         def study = new Study(
                 studyId: studyId,
                 secureObjectToken: isPublic ? Study.PUBLIC : studyId,
-                trialVisits: [] as Set,
                 dimensionDescriptions: [] as Set
         )
         study.save(flush: true, failOnError: true)
-        // add trial visits
+        // create trial visits
         if (trialVisitLabels) {
-            for (String label : trialVisitLabels) {
-                study.trialVisits.add(
+            for (String label: trialVisitLabels) {
+                trialVisits.add(
                         new org.transmartproject.db.i2b2data.TrialVisit(study: study, relTimeLabel: label)
                                 .save(flush: true, failOnError: true))
             }
         } else {
-            study.trialVisits.add(
+            trialVisits.add(
                     new org.transmartproject.db.i2b2data.TrialVisit(study: study, relTimeLabel: 'default')
                             .save(flush: true, failOnError: true))
         }
@@ -101,10 +101,10 @@ class AccessPolicyTestData extends org.transmartproject.rest.data.TestData {
         } as List<DimensionDescription>
         study.dimensionDescriptions.addAll(dimensionDescriptions)
         study.save(flush: true, failOnError: true)
-        study.trialVisits as List<TrialVisit>
+        trialVisits
     }
 
-    protected Concept createTestConcept(String conceptCode) {
+    protected static Concept createTestConcept(String conceptCode) {
         log.info "Creating test concept: ${conceptCode}"
         new ConceptDimension(conceptCode: conceptCode, conceptPath: "\\Test\\${conceptCode}")
                 .save(flush: true, failOnError: true)
@@ -175,7 +175,7 @@ class AccessPolicyTestData extends org.transmartproject.rest.data.TestData {
         patient.save(flush: true, failOnError: true)
     }
 
-    protected void createTestCategoricalObservations(
+    protected static void createTestCategoricalObservations(
             Patient patient, Concept concept, TrialVisit trialVisit, List<Map<String, String>> values, Date startDate) {
         log.info "Adding ${values?.size()} observations for patient: ${patient.subjectIds}, concept: ${concept.conceptCode}"
         int instanceNum = 1
@@ -197,7 +197,7 @@ class AccessPolicyTestData extends org.transmartproject.rest.data.TestData {
         }
     }
 
-    protected void createTestNumericalObservations(
+    protected static void createTestNumericalObservations(
             Patient patient, Concept concept, TrialVisit trialVisit, List<Map<String, BigDecimal>> values, Date startDate) {
         log.info "Adding ${values?.size()} observations for patient: ${patient.subjectIds}, concept: ${concept.conceptCode}"
         int instanceNum = 1

